Pocket/pocket-ios
# Share: update shortUrl on demand, if needed, when sharing an url

## Summary
* This PR fetches `shortUrl` if it's nil before sharing an item, in order to ensure we share `shortUrl` every time we can
## References 
* Branch name

## Implementation Details
* Add graphQl query to fetch an `Item`'s `shortUrl`
* Use it to update `shortUrl` if nil, when sharing an item: this can happen when users migrate from previous versions to `8.5.0`
## Test Steps
* Install the production (App Store) version and perform a refresh
* Within 5 minutes, install this branch as an upgrade (do not delete the prod version)
* Put a breakpoint where the share logic happens, for example [here](https://github.com/Pocket/pocket-ios/blob/5bcff04f2fac59e8197aaa6e893701d98e30542a/PocketKit/Sources/PocketKit/Home/HomeViewModel.swift#L799)
* verify that `shortUrl` is nil
* put another breakpoint [here](https://github.com/Pocket/pocket-ios/blob/5bcff04f2fac59e8197aaa6e893701d98e30542a/PocketKit/Sources/PocketKit/Home/HomeViewModel.swift#L806) and verify that the shortUrl has been fetched and assigned to the corresponding item
* verify that share works properly, and that you are actually sharing a `pocket.co` url
* 
## PR Checklist:
